EpiAware/src/EpiInfModels/RenewalSteps.jl

```diff
@@ -0,0 +1,125 @@
+@doc raw"
+Abstract type representing an accumulation iteration/step for a Renewal model with a constant
+generation interval.
+"
+abstract type AbstractConstantRenewalStep <: AbstractAccumulationStep end
+
+@doc raw"
+The renewal process iteration/step function struct with constant generation interval.
+
+Note that the generation interval is stored in reverse order.
+"
+struct ConstantRenewalStep{T} <: AbstractConstantRenewalStep
+    rev_gen_int::Vector{T}
+end
+
+@doc """
+    function (renewal_step::ConstantRenewalStep)(recent_incidence, Rt)
+
+Implement the `Renewal` model iteration/step function, with constant generation interval.
+
+## Mathematical specification
+
+The new incidence is given by
+
+```math
+I_t = R_t \\sum_{i=1}^{n-1} I_{t-i} g_i
+```
+
+where `I_t` is the new incidence, `R_t` is the reproduction number, `I_{t-i}` is the recent incidence
+and `g_i` is the generation interval.
+
+# Arguments
+- `recent_incidence`: Array of recent incidence values order least recent to most recent.
+- `Rt`: Reproduction number.
+
+# Returns
+- Updated incidence array.
+"""
+function (renewal_step::ConstantRenewalStep)(recent_incidence, Rt)
```

Comment:
My hope is we can do this just with the AR step in the future

Reply:
That would be very neat.

Reply:
It would be interesting to start reasoning on this. Both `AR`, `MA` and `Renewal` etc have/can be represented as having a convolution on recent variables. Otoh, the "noise" process is different.

Is this related to the compositional idea, e.g. one "Step" for doing a conv-like operation and another that specialises?

Reply:
I've renamed to `recurrent_step` following our discussion in #439